File: jiwon/031 하샤드 수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int num = 0;
    int xx = x;
    
    while(x != 0){
        num += x % 10;
        x /= 10; 
    }
    
    if(xx % num == 0){
        answer = true;
    } else{
        answer = false;
    }
    
    return answer;
}

File: lesunny/031_하샤드수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        int sum = 0;
        int tmpX = x;

        while(tmpX != 0){
            sum += (tmpX%10);
            tmpX /= 10;
        }
        
        int result = x%sum;
        
        if(result != 0)
            answer = false;
        
        return answer;
    }
}
